src/bin/elfshaker/pack.rs

```diff
@@ -134,6 +151,27 @@ pub(crate) fn run(matches: &ArgMatches) -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+fn packs_from_list(
+    repo: &Repository,
+    mut reader: impl io::Read,
+    separator: u8,
+) -> Result<Vec<PackId>, elfshaker::repo::Error> {
+    let mut buf = vec![];
+    reader.read_to_end(&mut buf)?;
+
+    buf.split(|c| *c == separator)
+        .filter(|s| !s.is_empty())
+        .map(std::str::from_utf8)
+        .map(|s| match s {
+            Ok(v) => repo.find_snapshot(v).map(|l| l.pack().clone()),
+            Err(e) => Err(elfshaker::repo::Error::IOError(io::Error::new(
+                io::ErrorKind::InvalidData,
+                e,
+            ))),
```

Comment:
I wonder if we can improve the error behaviour in this case- just thinking that if an unfortunate soul does have non-utf8, this error message might not help them understand the cause, since it looks like a repo io error and/or one that has to do with invalid date in the repo rather than in the snapshot list.